glean-core/src/upload/request.rs

```diff
@@ -29,11 +34,18 @@ impl PingRequest {
     /// Automatically creates the default request headers.
     /// Clients may add more headers such as `userAgent` to this list.
     pub fn new(document_id: &str, path: &str, body: JsonValue) -> Self {
+        // We want uploads to be gzip'd. Instead of doing this for each platform
+        // we have language bindings for, apply compression here.
+        let gzipped_content = Self::gzip_content(path, body.to_string().as_bytes());
+        let add_gzip_header = gzipped_content.is_some();
+        let body = gzipped_content.unwrap_or_else(|| body.to_string().as_bytes().to_vec());
```

Comment:
We can avoid one allocation:

```suggestion
        let body = gzipped_content.unwrap_or_else(|| body.to_string().into_bytes());
```

Reply:
Actually, we could avoid another allocation by having

```
let body = body.to_string();
```

above and use that in both in the gzip_content call and then here.